internal/store/model/templateversion.go

```diff
@@ -0,0 +1,112 @@
+package model
+
+import (
+	"encoding/json"
+	"time"
+
+	api "github.com/flightctl/flightctl/api/v1alpha1"
+	"github.com/flightctl/flightctl/internal/util"
+)
+
+var (
+	TemplateVersionAPI      = "v1alpha1"
+	TemplateVersionKind     = "TemplateVersion"
+	TemplateVersionListKind = "TemplateVersionList"
+)
+
+type TemplateVersion struct {
+	ResourceWithPrimaryKeyOwner
+
+	// The desired state, stored as opaque JSON object.
+	Spec *JSONField[api.TemplateVersionSpec]
+
+	// The last reported state, stored as opaque JSON object.
+	Status *JSONField[api.TemplateVersionStatus]
+}
+
+type TemplateVersionList []TemplateVersion
+
+func (t TemplateVersion) String() string {
+	val, _ := json.Marshal(t)
+	return string(val)
+}
+
+func NewTemplateVersionFromApiResource(resource *api.TemplateVersion) *TemplateVersion {
+	if resource == nil || resource.Metadata.Name == nil {
+		return &TemplateVersion{}
+	}
+
+	var status api.TemplateVersionStatus
+	if resource.Status != nil {
+		status = *resource.Status
+	}
+
+	return &TemplateVersion{
+		ResourceWithPrimaryKeyOwner: ResourceWithPrimaryKeyOwner{
+			Name:        *resource.Metadata.Name,
+			Generation:  resource.Metadata.Generation,
+			Owner:       resource.Metadata.Owner,
+			Annotations: util.LabelMapToArray(resource.Metadata.Annotations),
+		},
+		Spec:   MakeJSONField(resource.Spec),
+		Status: MakeJSONField(status),
+	}
+}
+
+func (t *TemplateVersion) ToApiResource() api.TemplateVersion {
+	if t == nil {
```

Comment:
can we add a comment on how this can be true

Reply:
Done